code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -817,3 +823,91 @@
 	actual_action.fix_borg(robot_target)
 	remove_ranged_ability(ranged_ability_user, "<span class='warning'>[robot_target] successfully rebooted.</span>")
 	return TRUE
+
+/datum/AI_Module/core_tilt
+	module_name = "Rolling Servos"
+	mod_pick_name = "watchforrollingcores"
+	description = "Allows you to slowly roll your core around, crushing anything in your path with your bulk."
+	cost = 10
+	one_purchase = FALSE
+	power_type = /datum/action/innate/ai/ranged/core_tilt
+	unlock_sound = 'sound/effects/bang.ogg'
+	unlock_text = "<span class='notice'>You gain the ability to roll over and crush anything in your way.</span>"
+
+/datum/action/innate/ai/ranged/core_tilt
+	name = "Roll Over"
+	button_icon_state = "roll_over"
+	desc = "Allows you to roll over in the direction of your choosing, crushing anything in your way."
+	auto_use_uses = FALSE
+	linked_ability_type = /obj/effect/proc_holder/ranged_ai/roll_over
+
+
+/obj/effect/proc_holder/ranged_ai/roll_over
+	active = FALSE
+	ranged_mousepointer = 'icons/effects/cult_target.dmi'
+	enable_text = "<span class='notice'>Your inner servos shift as you prepare to roll around. Click adjacent tiles to roll into them!</span>"
+	disable_text = "<span class='notice'>You disengage your rolling protocols.</span>"
+	COOLDOWN_DECLARE(time_til_next_tilt)
+	/// How long does it take us to roll?
+	var/roll_over_time = MALF_AI_ROLL_TIME
+	/// How long does it take for the ability to cool down, on top of [roll_over_time]?
+	var/roll_over_cooldown = MALF_AI_ROLL_COOLDOWN
+
+
+/obj/effect/proc_holder/ranged_ai/roll_over/InterceptClickOn(mob/living/caller, params, atom/target_atom)
+	if(..())
+		return
+	if(!isAI(ranged_ability_user))
+		return
```

Comment:
Surely this spell should only be given to AIs? 😛 Feels overprotective, but I dunno prevents admin shenanigans I guess.

Reply:
Always implement safeties for this sort of thing, admins love breaking stuff (source: me)